File: src/lib/__tests__/bilingual-name.test.ts
import { describe, it, expect } from "vitest";
import { readFileSync } from "node:fs";
import { join } from "node:path";
import { bilingualDisplay, bilingualName } from "../bilingual-name";

interface Case {
  id: string;
  locale: string;
  ko: string;
  en: string | null;
  roman: string | null;
  primary: string;
  secondary: string | null;
}

const fixture = JSON.parse(
  readFileSync(join(__dirname, "fixtures", "bilingual-name-cases.json"), "utf8"),
) as { cases: Case[] };

describe("bilingualName (공유 fixture ↔ Kit BilingualNameTests)", () => {
  it("fixture가 비어 있지 않다", () => {
    expect(fixture.cases.length).toBeGreaterThanOrEqual(10);
  });

  for (const c of fixture.cases) {
    it(c.id, () => {
      expect(bilingualName(c.locale, c.ko, { en: c.en, roman: c.roman })).toEqual({
        primary: c.primary,
        secondary: c.secondary,
      });
    });
  }

  it("bilingualDisplay는 한 줄 괄호이고 병기 없으면 primary만", () => {
    expect(bilingualDisplay({ primary: "Seolleung", secondary: "선릉역" })).toBe(
      "Seolleung (선릉역)",
    );
    expect(bilingualDisplay({ primary: "CU", secondary: null })).toBe("CU");
  });
});

File: android/kit/src/test/kotlin/space/dodoplanet/gildongmu/kit/BilingualNameTest.kt
package space.dodoplanet.gildongmu.kit

import kotlinx.serialization.Serializable
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertFalse
import kotlin.test.assertTrue

/** 웹 `bilingual-name.test.ts`·Kit `BilingualNameTests`와 같은 fixture — 규칙이 한 벌임을 강제. */
class BilingualNameTest {
    @Serializable
    private data class CaseFile(val cases: List<Case>) {
        @Serializable
        data class Case(
            val id: String, val locale: String, val ko: String, val en: String? = null, val roman: String? = null,
            val primary: String, val secondary: String? = null,
        )
    }

    private val cases by lazy { Fixtures.sharedJson("bilingual-name-cases.json", CaseFile.serializer()).cases }

    @Test fun fixtureIsNotEmpty() {
        assertTrue(cases.size >= 10)
    }

    @Test fun matchesSharedFixture() {
        for (c in cases) {
            val got = bilingualName(c.locale, c.ko, c.en, c.roman)
            assertEquals(c.primary, got.primary, "${c.id} primary")
            assertEquals(c.secondary, got.secondary, "${c.id} secondary")
        }
    }

    @Test fun displayIsOneLineParenthesis() {
        assertEquals("Seolleung (선릉역)", BilingualName("Seolleung", "선릉역").display)
        assertEquals("CU", BilingualName("CU", null).display)
    }

    @Test fun hasHangulMirrorsWeb() {
        assertTrue(hasHangul("강동성심병원"))
        assertTrue(hasHangul("ㄱ"))
        assertFalse(hasHangul("GS25"))
        assertFalse(hasHangul(""))
    }
}
